packages/eth-json-rpc-provider/src/safe-event-emitter-provider.ts

```diff
@@ -78,15 +76,10 @@ export class SafeEventEmitterProvider extends SafeEventEmitter {
    * @param eip1193Request - The request to send.
    * @returns The JSON-RPC response.
    */
-  async request<Params extends JsonRpcParams, Result extends Json>(
-    eip1193Request: Eip1193Request<Params>,
-  ): Promise<Result> {
+  async request(eip1193Request: Eip1193Request): Promise<Json> {
```

Comment:
Shouldn't `request` still be generic over `Params` and `Result`? This seems like a fundamental alteration of the EIP-1193 provider API that would make it unusable in many downstream call sites. We could check this using a preview build of this branch.